File: generate_frames.py
from PIL import Image, ImageDraw, ImageFont
import os

# Ensure directory exists
output_dir = "public/sequence"
if not os.path.exists(output_dir):
    os.makedirs(output_dir)

# Settings
width, height = 1920, 1080
bg_color = (18, 18, 18) # #121212
text_color = (255, 255, 255)
num_frames = 90

for i in range(num_frames):
    img = Image.new('RGB', (width, height), bg_color)
    d = ImageDraw.Draw(img)
    
    # Draw simple expansive circle based on frame
    radius = (i + 1) * 10
    d.ellipse((width//2 - radius, height//2 - radius, width//2 + radius, height//2 + radius), outline="cyan", width=5)
    
    # Draw Text
    text = f"Frame {i}"
    # d.text((width//2, height//2), text, fill=text_color) # simplistic positioning
    
    filename = f"frame_{i:02d}_delay-0.067s.webp"
    img.save(os.path.join(output_dir, filename), "WEBP")

print(f"Generated {num_frames} frames in {output_dir}")

File: generate_frames.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const outputDir = path.join(__dirname, 'public/sequence');
if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
}

async function generate() {
    for (let i = 0; i < 90; i++) {
        const svgImage = `
        <svg width="1920" height="1080" xmlns="http://www.w3.org/2000/svg">
            <rect width="100%" height="100%" fill="#121212"/>
            <circle cx="960" cy="540" r="${(i + 1) * 5}" stroke="cyan" stroke-width="5" fill="none"/>
            <text x="50%" y="50%" font-family="Arial" font-size="50" fill="white" text-anchor="middle" dy=".3em">Frame ${i}</text>
        </svg>
        `;

        const filename = `frame_${String(i).padStart(2, '0')}_delay-0.067s.webp`;
        await sharp(Buffer.from(svgImage))
            .webp()
            .toFile(path.join(outputDir, filename));

        console.log(`Generated ${filename}`);
    }
}

generate();
